lib/cmd/compile/styles.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+const _ = require('lodash'),
+  fs = require('fs-extra'),
+  h = require('highland'),
+  path = require('path'),
+  es = require('event-stream'),
+  gulp = require('gulp'),
+  rename = require('gulp-rename'),
+  changed = require('gulp-changed'),
+  cssmin = require('gulp-cssmin'),
+  gulpIf = require('gulp-if'),
+  detective = require('detective-postcss'),
+  autoprefixer = require('autoprefixer'),
+  postcss = require('gulp-postcss'),
+  cssImport = require('postcss-import'),
+  mixins = require('postcss-mixins'),
+  nested = require('postcss-nested'),
+  simpleVars  = require('postcss-simple-vars'),
+  helpers = require('../../compilation-helpers'),
+  componentsSrc = path.join(process.cwd(), 'styleguides', '**', 'components', '*.css'),
+  layoutsSrc = path.join(process.cwd(), 'styleguides', '**', 'layouts', '*.css'),
+  destPath = path.join(process.cwd(), 'public', 'css'),
+  variables = {
+    // asset host and path are set in different environments when using separate servers/subdomains for assets
+    'asset-host': process.env.CLAYCLI_COMPILE_ASSET_HOST ? process.env.CLAYCLI_COMPILE_ASSET_HOST.replace(/\/$/, '') : '',
+    'asset-path': process.env.CLAYCLI_COMPILE_ASSET_PATH || '',
+    // these arguments allow setting default env variables
+    minify:  process.env.CLAYCLI_COMPILE_MINIFIED || process.env.CLAYCLI_COMPILE_MINIFIED_STYLES || ''
+  };
+
+/**
+ * determine filepath for compiled css, based on the source filepath
+ * used to test if a css file has changed
+ * @param  {string} filepath
+ * @return {string}
+ */
+function transformPath(filepath) {
+  const component = path.basename(filepath, '.css'), // component name, plus variation if applicable
+    pathArray = path.dirname(filepath).split(path.sep),
+    styleguide = pathArray[pathArray.length - 2]; // parses 'styleguides/<styleguide>/components' for the name of the styleguide
+
+  return path.join(destPath, `${component}.${styleguide}.css`);
+}
+
+/**
+ * determine if a file (or its dependencies) has changed
+ * note: this only checks ONE level of dependencies, because that's all we allow in styleguide css
```

Comment:
Where is that mentioned? I know its an `nymag` convention to do that, but if that's the case we should specify in the readme section for `clay compile styles`.  That would be a huge pit if you used two levels of CSS imports and couldn't figure out why your css wasn't updating unless you blew out your `public` directory.

Reply:
good point